src/mono/mono/mini/aot-compiler.c

```diff
@@ -4331,6 +4331,12 @@ collect_dedup_method (MonoAotCompile *acfg, MonoMethod *method)
 static int
 add_method_full (MonoAotCompile *acfg, MonoMethod *method, gboolean extra, int depth)
 {
+	if (collect_dedup_method (acfg, method))
```

Comment:
What is the purpose of this change ?

Reply:
Before this change, during the runtime, `(wrapper runtime-invoke) void object:runtime_invoke_dynamic (intptr,intptr,intptr,intptr)` was being searched in the dedup AOT module as `RUNTIME_INVOKE_WRAPPER`, but it was not emitted in the dedup AOT module.

Dynamic runtime invoke wrapper has been added to the list of methods which need to be deduplicated by invoking `add_method_full`. Moving `collect_dedup_method` to `add_method_full` method deduplicates `runtime_invoke_dynamic` that has been added as below.
https://github.com/dotnet/runtime/blob/aaa1de18a7f97774fd44a1728dafc0d86894800e/src/mono/mono/mini/aot-compiler.c#L4864-L4867